module/Photo/src/Photo/Service/AlbumCover.php

```diff
@@ -0,0 +1,158 @@
+<?php
+
+namespace Photo\Service;
+
+use Application\Service\AbstractService;
+use Zend\ServiceManager\ServiceManager;
+use Zend\ServiceManager\ServiceManagerAwareInterface;
+use Photo\Model\Album as AlbumModel;
+use Photo\Model\Photo as PhotoModel;
+use Imagick;
+
+/**
+ * Album cover services. Used for (re)generating album covers.
+ * 
+ */
+class AlbumCover extends AbstractService
+{
+
+    public function createCover($album)
+    {
+        $cover = $this->generateCover($album);
+        $tempFileName = sys_get_temp_dir() . '/CoverImage' . rand() . '.png';
+        $cover->writeImage($tempFileName);
+        $newPath = $this->getPhotoService()->generateStoragePath($tempFileName);
+        $config = $this->getConfig();
+        rename($tempFileName, $config['upload_dir'] . '/' . $newPath);
+        return $newPath;
+    }
+
+    protected function generateCover($album)
```

Comment:
This method needs some clarificatoin. What exactly happens, and why is it neede?